### PR TITLE
[zh-cn]: Update translation of String.match()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13975

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 1]
```diff
@@ -5,14 +5,14 @@ slug: Web/JavaScript/Reference/Global_Objects/String/match
 
 {{JSRef}}
 
-**`match()`** 方法检索返回一个字符串匹配正则表达式的结果。
+**`match()`** 方法检索字符串与[正则表达式](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions)的匹配结果。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:8]:
```suggestion
**`match()`** 方法检索字符串与[正则表达式](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions)进行匹配的结果。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 2]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
 
 ### 返回值
 
-- 如果使用 g 标志，则将返回与完整正则表达式匹配的所有结果，但不会返回捕获组。
-- 如果未使用 g 标志，则仅返回第一个完整匹配及其相关的捕获组（`Array`）。在这种情况下，返回的项目将具有如下所述的其他属性。
+一个 {{jsxref("Array")}}，其内容取决于全局 (`g`) 标志的存在或缺失，如果没有匹配，则返回 [`null`](/zh-CN/docs/Web/JavaScript/Reference/Operators/null)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:29]:
```suggestion
一个 {{jsxref("Array")}}，其内容取决于是否存在全局（`g`）标，如果没有匹配，则返回 [`null`](/zh-CN/docs/Web/JavaScript/Reference/Operators/null)。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 3]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+- 如果使用 `g` 标志，则将返回与完整正则表达式匹配的所有结果，但不会返回捕获组。
+- 如果没有使用 `g` 标志，则只返回第一个完整匹配及其相关捕获组。在这种情况下，`match()` 方法将返回与 {{jsxref("RegExp.prototype.exec()")}} 相同的结果（一个带有一些额外属性的数组）。
 
 ## 描述
 
-如果正则表达式不包含 `g` 标志，`str.match()` 将返回与 {{jsxref("RegExp.prototype.exec()", "RegExp.exec()")}}. 相同的结果。
+`String.prototype.match` 方法本身的实现非常简单，它只是使用字符串作为第一个参数调用了参数的 `Symbol.match` 方法。实际的实现来自于 [`RegExp.prototype[@@match]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match)。
 
-### 参看：`RegExp` 方法
+- 如果你需要知道一个字符串是否与一个正则表达式 {{jsxref("RegExp")}} 匹配，使用 {{jsxref("RegExp.prototype.test()")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:38]:
```suggestion
- 如果你需要知道一个字符串是否与一个正则表达式 {{jsxref("RegExp")}} 匹配，请使用 {{jsxref("RegExp.prototype.test()")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 4]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
-- 如果你只是需要第一个匹配结果，你也可以使用 {{jsxref("RegExp.prototype.exec()", "RegExp.exec()")}} 。
-- 如果你想要获得捕获组，并且设置了全局标志，你需要用 {{jsxref("RegExp.prototype.exec()", "RegExp.exec()")}} 或者 {{jsxref("String.prototype.matchAll()", " String.prototype.matchAll()")}}
+有关传入正则表达式时 `match()` 方法的语义更多信息，请参阅 [`RegExp.prototype[@@match]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@match)。
 
 ## 示例
 
-### 示例：使用 `match`
+### 使用 match()
 
 在下例中，使用 `match` 查找 "`Chapter`" 紧跟着 1 个或多个数值字符，再紧跟着一个小数点和数值字符 0 次或多次。正则表达式包含 `i` 标志，因此大小写会被忽略。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:48]:
```suggestion
在下例中，使用 `match` 查找 `"Chapter"` 紧跟着 1 个或多个数值字符，再紧跟着一个小数点和数值字符 0 次或多次。正则表达式包含 `i` 标志，因此大小写会被忽略。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 5]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+//   input: 'For more information, see Chapter 3.4.5.1',
+//   groups: undefined
+// ]
 ```
 
-### 示例：`match` 使用全局（global）和忽略大小写（ignore case）标志
+在上面的匹配结果中，`'see Chapter 3.4.5.1'` 是整个匹配。`'Chapter 3.4.5.1'` 被 `(chapter \d+(\.\d)*)` 捕获。`.1` 是由 `(\.\d)` 最后捕获的值。`index` 属性 (`22`) 是整个匹配的零基索引。`input` 属性是解析的原始字符串。
 
-下例展示了 `match` 使用 global 和 ignore case 标志。A-E、a-e 的所有字母将会作为一个数组的元素返回。
+### 使用全局标志和忽略大小写标志掉用 match()
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:68]:
```suggestion
### 将全局标志和忽略大小写标志与 match() 一起使用
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 6]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+
+下面的示例演示了在 `match()` 中使用全局标志和忽略大小写标志。所有字母 `A` 到 `E` 和 `a` 到 `e` 都将作为数组中的一个元素返回。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:70]:
```suggestion
下面的示例演示了在 `match()` 中使用全局标志和忽略大小写标志。所有从 `A` 到 `E` 和从 `a` 到 `e` 的字母都将作为数组中的一个元素返回。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 7]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+const regexp = /[A-E]/gi;
+const matches = str.match(regexp);
 
-console.log(matches_array);
+console.log(matches);
 // ['A', 'B', 'C', 'D', 'E', 'a', 'b', 'c', 'd', 'e']
 ```
 
-### 使用 `match()`，不传参数
+> **备注：** 还可以参见 {{jsxref("String.prototype.matchAll()")}} 和[通过标志进行高级搜索](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions#通过标志进行高级搜索)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:81]:
```suggestion
> **备注：** 参见 {{jsxref("String.prototype.matchAll()")}} 和[通过标志进行高级搜索](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions#通过标志进行高级搜索)。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 8]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
 
 ```js
-var str = "Nothing will come of nothing.";
+const paragraph = "The quick brown fox jumps over the lazy dog. It barked.";
 
-str.match();   // returns [""]
+const capturingRegex = /(?<animal>fox|cat) jumps over/;
+const found = paragraph.match(capturingRegex);
+console.log(found.groups); // {animal: "fox"}
+
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:93]:
````suggestion
```

````

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 9]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+### 使用 match()，不传参数
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:94]:
```suggestion
### 不传参数使用 match()
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 10]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+如果一个对象有一个名为 `Symbol.match` 的方法，那么它可以被用作自定义匹配器。`Symbol.match` 的返回值将成为 `match()` 方法的返回值。
+
+```js
+const str = "Hmm, this is interesting.";
+
+str.match({
+  [Symbol.match](str) {
+    return ["Yes, it's interesting."];
+  },
+}); // returns ["Yes, it's interesting."]
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:113]: please translate this

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 11]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
 ```
 
 ### 一个非正则表达式对象作为参数
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:116]:
```suggestion
### 将非正则表达式对象作为参数
```

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 12]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
 
-当参数是一个字符串或一个数字，它会使用 new RegExp(obj) 来隐式转换成一个 {{jsxref("RegExp")}}。如果它是一个有正号的正数，RegExp() 方法将忽略正号。
+当 `regexp` 参数是一个字符串或一个数字，它会使用 `new RegExp(regexp)` 来隐式转换成一个 {{jsxref("RegExp")}}。
+
+```js
+const str1 =
+  "NaN 表示不是一个数字。在 JavaScript 中 Infinity 包括了 -Infinity 和 +Infinity。";
+const str2 =
+  "我的爷爷已经 65 岁了，我的奶奶已经 63 岁";
+const str3 = "该合同被声明为 null 且 void。";
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:125]:
```suggestion
const str3 = "该合同被宣布无效。";
```

Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:125]: 下文需要在 str3 中找到 null

File: files/zh-cn/web/javascript/reference/global_objects/string/match/index.md [hunk 13]
```diff
@@ -26,92 +26,123 @@ str.match(regexp)
...
+str1.match("数字"); // “数字” 是一个字符串。返回 ["number"]
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/match/index.md:126]:
```suggestion
str1.match("数字"); // “数字” 是一个字符串。返回 ["数字"]
```

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t

**Pull request merged**